roll/__init__.py

```diff
@@ -300,91 +345,201 @@ def cookies(self):
             self._cookies = self.app.Cookies()
         return self._cookies
 
+    def __bytes__(self):
+        response = b'HTTP/1.1 %a %b\r\n' % (
+            self.status.value, self.status.phrase.encode())
+
+        if self._cookies:
+            # https://tools.ietf.org/html/rfc7230#page-23
+            for cookie in self.cookies.values():
+                response += b'Set-Cookie: %b\r\n' % str(cookie).encode()
+
+        # https://tools.ietf.org/html/rfc7230#section-3.3.2 :scream:
+        for key, value in self.headers.items():
+            response += b'%b: %b\r\n' % (key.encode(), str(value).encode())
+
+        if not self.bodyless:
+            if not isinstance(self.body, bytes):
+                body = str(self.body).encode()
+            else:
+                body = self.body
+
+            if 'Content-Length' not in self.headers:
+                response += b'Content-Length: %i\r\n' % len(body)
+
+            response += b'\r\n'
+            if body:
+                response += body
+        else:
+            response += b'\r\n'
+        return response
+
+
+# PROTOCOL STATUS FLAGS
```

Comment:
This probably needs a documentation, at least to understand why we jump from 2 to 4. Reading documentation from https://developer.mozilla.org/en-US/docs/Web/HTTP/Protocol_upgrade_mechanism I don't get why we have to do that.

Reply:
I'm ashamed to say that it's yet again a left over from when they were used as bit masks...
No reason !

Reply:
No shame, that's why code reviews are so important 😉